presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -220,7 +222,7 @@ private <T extends PrestoSparkTaskOutput> JavaPairRDD<MutablePartitionId, T> cre
                 outputType);
 
         Optional<PrestoSparkTaskSourceRdd> taskSourceRdd;
-        List<TableScanNode> tableScans = findTableScanNodes(fragment.getRoot());
+        List<PrestoSparkTaskSource> tableScans = findTableScanNodes(fragment.getRoot());
```

Comment:
not really a big fan calling the new class `PrestoSparkTaskSource`. `Task` is a physical concept. Shall we call it `PrestoSparkSource`? Also would be good to rename `tableScans` into `sources`